attorney_online/include/datatypes.h

```diff
@@ -0,0 +1,541 @@
+   *
+   * <https://github.com/AttorneyOnline/docs/blob/master/docs/development/network.md#in-character-message>
+   */
+  enum class DESKMOD {
+    CHAT = -1,  //!< Mostly deprecated. Falls back to hardcoded table behaviour.
+    HIDDEN = 0, //!< The desk is fully hidden.
+    SHOWN = 1, //!< The desk is shown in front of character and other_character.
+    HIDEPREANIM = 2, //!< The desk is hidden during preanim, shown when it ends
+    SHOWPREANIM = 3, //!< The desk is shown during preanim, hidden when it ends
+    HIDEPREANIM_PAIR = 4,  //!< The desk is hidden during preanim, character is
+                           //!< centered and pairing is ignored, when it ends
+                           //!< desk is shown and pairing is restored
+    SHOWPREANIM_NOPAIR = 5 //!< The desk is shown during preanim, when it ends
+                           //!< character is centered and pairing is ignored
+  };
+  Q_ENUM(DESKMOD)
+
+  /**
+   * @brief Controls wether special emote effects are used.
+   *
+   * <https://github.com/AttorneyOnline/docs/blob/master/docs/development/network.md#in-character-message>
+   */
+  enum class EMOTE_MODIFIER {
+    NONE =
+        0,   //!< No preanimation, Overridden to 2 by a non-0 objection modifier
+    PREANIM, //!< Play preanimation and SFX.
+    OBJECTION,  //!< Play objection, then preanimation.
+    ZOOM = 5,   //!< No preanimation and apply zoom background.
+    OBJECT_ZOOM //!< Play objection and apply zoom background after.
+  };
+  Q_ENUM(EMOTE_MODIFIER)
+
+  /**
+   * @brief Controls which shout is used.
+   *
+   * @details In typical AO fashion the custom one is bundled with the filename
+   * in the same argument. Make sure to split the & delimiter away and only pass
+   * the integer when checking for the enum.
+   *
+   * <https://github.com/AttorneyOnline/docs/blob/master/docs/development/network.md#in-character-message>
+   */
+  enum class SHOUT_MODIFIER {
+    NOTHING = 0,
+    HOLD_IT,
+    OBJECTION,
+    TAKE_THAT,
+    CUSTOM
+  };
+  Q_ENUM(SHOUT_MODIFIER)
+
+  /**
+   * @brief Controls which color a text message appears in.
+   *
+   * @details Prior to 2.8 [CITATION NEEDED!] this is the only way to change the
+   * color of your message, but it has been made largely obsolete by inline text
+   * color, though still serves some purpose.
+   *
+   * Rainbow text has been remove in 2.8 and will just appear as plain white.
+   * Sorry folks, no pride in AO.
+   *
+   */
+  enum class TEXT_COLOR {
+    WHITE = 0,
+    GREEN,
+    RED,
+    ORANGE,
+    BLUE,
+    YELLOW,
+    RAINBOW //!< DEPRECATED! Removed in 2.8!
+  };
+  Q_ENUM(TEXT_COLOR)
+
+  enum class AUTHENTICATION {
+    LOGOUT = -1,  //!< Log-out. Hides the guard button.
+    LOGIN_FAIL,   //!< Unsuccessful log-in attempt.
+    LOGIN_SUCCESS //!< Successful log-in. Displays the guard button.
+  };
+  Q_ENUM(AUTHENTICATION)
+
+  /**
+   * @brief The TIMER_ACTION enum
+   *
+   *  @details Allan, please add details.
```

Comment:
who is allan

Reply:
![image](https://user-images.githubusercontent.com/62221668/208360528-9eb8d49e-f202-4610-9e65-f5ba20acf8d5.png)

I admit, I got lazy and did not document these as briefly as "at glance" understanding of AOs network without resorting to opening network.md requires. This was supposed to be a reminder to do it, which I did not. 

TLDR : It turns out I am Allan.